### ISSUE TITLE
Issues with on/off target log ratios

### ISSUE THREAD

**Issue by waemm** [lima1/PureCN#156]
Hi Markus,

I've had some strange results out of a recent batch of samples. It appears 3/5 samples have 

Mean standard deviation of on-target log-ratios only: 0.0
Mean standard deviation of off-target log-ratios only: 0.0

I include two example log files. One of these samples performs pretty poorly, having very few believable somatic variants - several of these are also flagged by PureCN. 

Some questions.
1. Have you seen this before, what could be causing this?
2. Are there any avenues I should investigate regarding the raw data to ensure something hasn't gone wrong during processing?
3. With flagged variants, I believe in my case this can only be due to potential presence in the mapping bias file as I don't use a normalDB (as this is already done in cnvkit), is it worth attempting to recover some of these variants or are they very likely to be spurious?

Thanks,
Warren
[sample1.log](https://github.com/lima1/PureCN/files/5660076/sample1.log)
[sample2.log](https://github.com/lima1/PureCN/files/5660077/sample2.log)

**Comment by lima1** [comment]
Hi Warren,

in the CNVkit import, PureCN will look at the CNVkit log2 ratio in the tumor file. If this looks super noisy, it will ignore it and simply replace them with the one provided in the segmentation. Since those do not change within the segments by definition, the standard deviation is 0 (PureCN tries to estimate the within segment standard deviation).

So likely  an issue upstream, you can plot the log2 ratios and see if it looks weird.

Not sure the flagging is super reliable in this case.

Best,
Markus

**Comment by waemm** [comment]
Thanks for the prompt reply Markus, I have taken a look and it doesn't seem too clear cut. From one batch two of the samples it looks like i can explain this quite clearly as there is clearly an artefact. The final sample (ending 659) doesn't seem to have a massive shift and yet still has SD 0.  (![Untitled 3](https://user-images.githubusercontent.com/71687/101760111-56e01000-3ad2-11eb-9d1b-f657f73b5208.png)

The samples that are not considered noisy are here: 
![good_samples](https://user-images.githubusercontent.com/71687/101761722-682a1c00-3ad4-11eb-8f45-05b67c79a7ff.png)

For another batch it is also not clear cut, I get one sample that has 0 SD (marked with an X) and the others all have SD values even though one of them looks very much like the noisy one. Any thoughts?
![example2](https://user-images.githubusercontent.com/71687/101761964-b63f1f80-3ad4-11eb-8a32-184822e29ae4.png)


Thanks for your help!

**Comment by lima1** [comment]
Hmm, if you remove all input lines with log-ratio < -8, does it work? You can also set the sex to M to remove chrom X and Y.

**Comment by waemm** [comment]
Thanks Markus, I gave this a try, it looks like setting sex to M doesn't solve the problem even though the vast majority are coming from chrY (351/364)  . Based on the data there are ~10 segments from non-sex chromosomes that have log2 < -20 (the two samples that struggle both look like this) that are throwing off the data. Is there no way to ignore these values or must this be corrected manually in the CNR file?

**Comment by lima1** [comment]
Hi Warren, I do ignore them, but probably not until after that step. I'll have a look later today.

**Comment by waemm** [comment]
Hi MArkus, sorry for taking so long to get back to you. With this really problematic sample I have removed all log2 < -8 from and I still get the same. This indicates to me some underlying issue with the data, can you reccomend a way I can explore this further? Could this be due to issues with sample collection or processing ?

INFO [2021-01-22 11:10:18] Mean standard deviation of on-target log-ratios only: 0.00
INFO [2021-01-22 11:10:18] Mean standard deviation of off-target log-ratios only: 0.00

**Comment by lima1** [comment]
Hi Warren. Can you share the coverage file of this particular sample? Feel free to send by email if not publicly.

**Comment by waemm** [comment]
Thanks Markus, have sent you an email.

**Comment by waemm** [comment]
Hi Markus, just checking in if you've had a chance to look at the sample?

**Comment by lima1** [comment]
Sorry Warren, I'm kind of swamped right now with other projects, but will find some time in the next days. Maybe I can encourage you to try our own normalization plus the PSCBC segmentation and compare against CNVkit? Would be curious to see how it performs.

**Comment by waemm** [comment]
Hi Markus, apologies, please do take your time. I'll see what i can do to run this directly through purecn.

**Comment by lima1** [comment]
Closing now, looks like the sample indeed has a very noisy log-ratio stored and  the ratios are correctly ignored.
